ct_optcon/examples/NLOC_generalConstrained.cpp

```diff
@@ -74,66 +69,58 @@ class PointMass : public ct::core::ControlledSystem<STATE_DIM, CONTROL_DIM> {
  * are represented by a circle.
  */
 
- // class representing a general constraint that ensures obstacle avoidance
-class ObstacleConstraintSimple
-        : public ct::optcon::ConstraintBase<STATE_DIM, CONTROL_DIM> {
+// class representing a general constraint that ensures obstacle avoidance
+class ObstacleConstraintSimple : public ct::optcon::ConstraintBase<STATE_DIM, CONTROL_DIM>
+{
 public:
     EIGEN_MAKE_ALIGNED_OPERATOR_NEW
     typedef ct::optcon::ConstraintBase<STATE_DIM, CONTROL_DIM> Base;
     typedef ct::core::StateVector<STATE_DIM> state_vector_t;
     typedef ct::core::ControlVector<CONTROL_DIM> control_vector_t;
     typedef Eigen::Matrix<double, 2, 1> Vector2s;
 
-    ObstacleConstraintSimple(const Vector2s x0, const double r) : x0_(x0), r_(r) {
+    ObstacleConstraintSimple(const Vector2s x0, const double r) : x0_(x0), r_(r)
+    {
         Base::lb_.resize(1);
         Base::ub_.resize(1);
         Base::lb_.setConstant(0.0);
         Base::ub_.setConstant(std::numeric_limits<double>::max());
     }
 
-    ObstacleConstraintSimple(const ObstacleConstraintSimple &other)
-            : Base(other), x0_(other.x0_) {
+    ObstacleConstraintSimple(const ObstacleConstraintSimple& other)
+    {
         Base::lb_.resize(1);
         Base::ub_.resize(1);
         Base::lb_.setConstant(0.0);
         Base::ub_.setConstant(std::numeric_limits<double>::max());
     }
 
-    virtual ~ObstacleConstraintSimple() = default;
-
-    virtual ObstacleConstraintSimple *clone() const override {
-        return new ObstacleConstraintSimple(*this);
-    }
-
+    virtual ObstacleConstraintSimple* clone() const override { return new ObstacleConstraintSimple(*this); }
     virtual size_t getConstraintSize() const override { return 1; }
-
-    virtual Eigen::VectorXd evaluate(const state_vector_t &x,
-                                     const control_vector_t &u,
-                                     const double t) override {
+    virtual Eigen::VectorXd evaluate(const state_vector_t& x, const control_vector_t& u, const double t) override
+    {
         Eigen::Matrix<double, 1, 1> val;
         // returns a number > 0 if point x is outside of the ellipsoid
-        val.template segment<1>(0) << (1/pow(r_,2)) * (x(0) - x0_(0)) * (x(0) - x0_(0)) +
-                                      (1/pow(r_,2)) * (x(1) - x0_(1)) * (x(1) - x0_(1)) -
-                                      double(1.0);
+        val.template segment<1>(0) << (1 / pow(r_, 2)) * (x(0) - x0_(0)) * (x(0) - x0_(0)) +
+                                          (1 / pow(r_, 2)) * (x(1) - x0_(1)) * (x(1) - x0_(1)) - double(1.0);
```

Comment:
@hmcm thanks for addressing the comments! I have one more small question: could you quickly verify that the obstacle constraint equation is correct? There is a member variable `r_` that you set to 1.5, but when I look at the final plot, it seems like the curve is much "bigger" than r=1.5. Could you quickly counter-check that the equations here are correct?

Reply:
@markusgft I double-checked and I think the equation is correct. I agree with you that the curve looks a bit counter-intuitive though. However I am not sure what the cause of this is. For now, I've chosen a time shorter horizon which makes the trajectory more intuitive. I have added the obstacle to the plot to illustrate the results a little better.